tests/test_filesystem.py

```diff
@@ -61,26 +85,761 @@ def test_win_devices_not_win(self):
+    # Cut down a bit on the waiting time
+    @set_config({"wait_ext_drive": 1})
+    def test_dir_nonexistent_darwin(self):
+        # Within /Volumes
+        assert filesystem.check_mount("/Volumes/nosuchdir") is False  # Issue #1457
+        assert filesystem.check_mount("/Volumes/noSuchDir/") is False
+        assert filesystem.check_mount("/Volumes/nosuchDIR/subdir") is False
+        assert filesystem.check_mount("/Volumes/NOsuchdir/subdir/") is False
+
+    @set_platform("darwin")
+    def test_dir_outsider_darwin(self):
+        # Outside of /Volumes
+        assert filesystem.check_mount("/test/that/") is True
+        # Root directory
+        assert filesystem.check_mount("/") is True
+
+
+class TestCheckMountWin(ffs.TestCase):
+    # Our faked windows directory
+    test_dir = r"F:\test\dir"
+
+    def setUp(self):
+        self.setUpPyfakefs()
+        self.fs.is_windows_fs = True
+        self.fs.is_case_sensitive = False
+        self.fs.path_separator = "\\"
+        self.fs.create_dir(self.test_dir)
+        # Sanity check the fake filesystem
+        assert os.path.exists(self.test_dir) is True
+
+    @set_platform("win32")
+    def test_existing_dir_win(self):
+        assert filesystem.check_mount("F:\\test") is True
+        assert filesystem.check_mount("F:\\test\\dir\\") is True
+        # Filesystem and drive letters are case-insensitive on this platform
+        assert filesystem.check_mount("f:\\Test") is True
+        assert filesystem.check_mount("f:\\test\\DIR\\") is True
+
+    @set_platform("win32")
+    def test_bare_mountpoint_win(self):
+        assert filesystem.check_mount("F:\\") is True
+        assert filesystem.check_mount("Z:\\") is False
+
+    @set_platform("win32")
+    def test_dir_nonexistent_win(self):
+        # The existance of the drive letter is what really matters
+        assert filesystem.check_mount("F:\\NoSuchDir") is True
+        assert filesystem.check_mount("F:\\NoSuchDir\\") is True
+        assert filesystem.check_mount("F:\\NOsuchdir\\subdir") is True
+        assert filesystem.check_mount("F:\\nosuchDIR\\subdir\\") is True
+
+    @set_platform("win32")
+    # Cut down a bit on the waiting time
+    @set_config({"wait_ext_drive": 1})
+    def test_dir_on_nonexistent_drive_win(self):
+        # Non-existent drive-letter
+        assert filesystem.check_mount("H:\\NoSuchDir") is False
+        assert filesystem.check_mount("E:\\NoSuchDir\\") is False
+        assert filesystem.check_mount("L:\\NOsuchdir\\subdir") is False
+        assert filesystem.check_mount("L:\\nosuchDIR\\subdir\\") is False
+
+    @set_platform("win32")
+    def test_dir_outsider_win(self):
+        # Outside the local filesystem
+        assert filesystem.check_mount("//test/that/") is True
+
+
+class TestTrimWinPath:
+    @set_platform("win32")
+    def test_short_path(self):
+        assert filesystem.trim_win_path(r"C:\short\path") == r"C:\short\path"
+
+    # @set_platform("win32")
```

Comment:
Great find, we need to fix it.
Let's instead mark it `@pytest.mark.xfail(sys.platform == "win32", reason="Bug in trim_win_path")`

Reply:
Done